File: BlogApp-RN/src/features/posts/domain/usecases/getComments.ts
import { AsyncResult } from "../../../../core/typedefs/asyncResult";
import { CompletedComment } from "../../data/models/completedComment";
import { PostDetails } from "../../data/models/postDetails";
import PostRepo from "../repos/postRepo";

class GetComments {
  constructor(private repo: PostRepo) {}
  call: (postId: string) => AsyncResult<CompletedComment[]> = (postId) => {
    return this.repo.getCommentsByPostId(postId);
  };
}

export default GetComments;

File: blogapp_flutter/lib/features/posts/domain/usecases/get_comments.dart
import 'package:blogapp_flutter/core/typedefs/async_result.dart';
import 'package:blogapp_flutter/core/typedefs/usecases.dart';
import 'package:blogapp_flutter/features/posts/data/models/completed_comment.dart';
import 'package:blogapp_flutter/features/posts/domain/repos/posts_repo.dart';

class GetComments extends UsecaseWithParams<List<CompletedComment>, String> {
  final PostsRepo _repo;
  GetComments(this._repo);
  @override
  AsyncResult<List<CompletedComment>> call(params) {
    return _repo.getCommentsByPostId(params);
  }
}
